projects/ad4110/src/app/main.c

```diff
@@ -0,0 +1,145 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "irq.h"
+#include "irq_extra.h"
+#include "gpio_irq_extra.h"
+#include "spi.h"
+#include "spi_extra.h"
+#include "ad4110.h"
+#include "error.h"
+#include "xparameters.h"
+#include <stdint.h>
+
+#define SPI_DEVICE_ID 	0U
+#define GPIO_OFFSET  	31U
+#define NREADY_PIN   	GPIO_OFFSET + 1U
+#define BUF_LENGTH 		32U
+#define GPIO_IRQ_ID 	52U
+
+int32_t data_buf[BUF_LENGTH];
+
+/***************************************************************************//**
+ * @brief main
+*******************************************************************************/
+int main()
+{
+	uint32_t ret;
+
+	struct xil_spi_init_param spi_extra = {
+		.type = SPI_PS,
+		.flags = 0U
+	};
+	struct spi_init_param spi_ip = {
+		.device_id = SPI_DEVICE_ID,
+		.max_speed_hz = 100000,
+		.mode = SPI_MODE_0,
+		.chip_select = 0U,
+		.bit_order = SPI_BIT_ORDER_MSB_FIRST,
+		.platform_ops = &xil_spi_ops,
+		.extra = &spi_extra
+	};
+
+	/* IRQ instance. */
+	struct irq_ctrl_desc *irq_desc;
+	struct xil_irq_init_param irq_extra = {
+		.type = IRQ_PS
+	};
+	struct irq_init_param irq_ip = {
+		.irq_ctrl_id = 0U,
+		.platform_ops = &xil_irq_ops,
+		.extra = &irq_extra
+	};
+
+	ret = irq_ctrl_init(&irq_desc, &irq_ip);
+	if(ret)
+		return FAILURE;
+	ret = irq_global_enable(irq_desc);
+	if(ret)
+		return FAILURE;
+
+	/* GPIO IRQ instance. */
+	struct irq_ctrl_desc *gpio_irq_desc;
+	struct xil_gpio_irq_init_param gpio_irq_extra = {
+		.parent_desc = irq_desc,
+		.gpio_device_id = XPAR_PS7_GPIO_0_DEVICE_ID,
+		.retriggerable = false
+	};
+	struct irq_init_param gpio_irq_ip = {
+		.irq_ctrl_id = GPIO_IRQ_ID,
+		.platform_ops = &xil_gpio_irq_ops,
+		.extra = &gpio_irq_extra
+	};
+
+	ret = irq_ctrl_init(&gpio_irq_desc, &gpio_irq_ip);
+	if(ret)
+		return FAILURE;
+
+	/* Device AD4110 instance. */
+	struct ad4110_dev *ad4110_dev;
+	struct ad4110_init_param ad4110_dev_ip = {
+		.spi_init = spi_ip,
+		.volt_ref = AD4110_INT_2_5V_REF,
+		.data_stat = AD4110_ENABLE,
+		.data_length = AD4110_DATA_WL24,
+		.afe_crc_en = AD4110_AFE_CRC,
+		.adc_crc_en = AD4110_ADC_CRC_CRC,
+		.op_mode = AD4110_VOLTAGE_MODE,
+		.gain = AD4110_GAIN_0_5,
+		.sync = AD4110_SYNC_DIS,
+		.afe_clk = AD4110_AFE_ADC_CLOCKED,
+		.adc_clk = AD4110_ADC_INT_CLK_CLKIO,
+		.addr = 0U,
+		.irq_desc = gpio_irq_desc,
+		.nready_pin = NREADY_PIN
+	};
+
+	ret = ad4110_setup(&ad4110_dev, ad4110_dev_ip);
+	if(ret)
```

Comment:
some prints would be nice also :). both in case of successful initialization and failure.

Reply:
Got it. Thanks!